src/riak_dt_lwwreg.erl

```diff
@@ -78,23 +78,16 @@ value(timestamp, {_V, TS}) ->
 
 %% @doc Assign a `Value' to the `lwwreg()'
 %% associating the update with time `TS'
--spec update(lwwreg_op(), term(), lwwreg()) ->
-                    {ok, lwwreg()}.
+-spec update(lwwreg_op(), term(), lwwreg()) -> {ok, lwwreg()}.
 update({assign, Value, TS}, _Actor, {_OldVal, OldTS}) when is_integer(TS), TS > 0, TS >= OldTS ->
     {ok, {Value, TS}};
 update({assign, _Value, _TS}, _Actor, OldLWWReg) ->
     {ok, OldLWWReg};
 %% For when users don't provide timestamps
 %% don't think it is a good idea to mix server and client timestamps
-update({assign, Value}, _Actor, {OldVal, OldTS}) ->
+update({assign, Value}, Actor, OldLWWReg) ->
     MicroEpoch = make_micro_epoch(),
-    LWW = case MicroEpoch > OldTS of
-              true ->
-                  {Value, MicroEpoch};
-              false ->
-                  {OldVal, OldTS}
-          end,
-    {ok, LWW}.
+    update({assign, Value, MicroEpoch}, Actor, OldLWWReg).
```

Comment:
That reads like an elegant solution.

The server now adds a timestamp, one could say. If the user would have written a timestamp before with a computer in a time zone ahead of server (14:00 their time) and then without a timestamp at 14:30, but that's 13:30 server time, then the value will not be overwritten.
But that is exactly compatible with what would have happened before, so that does not at all change the behaviour. The only difference is this little possibility of the same non time-stamped message appears twice at exactly the same micro epoch. Before we would not update the value, now we do... the last write wins (or the write taking the longest time to arrive). That seems fair.